Can't install 1.0.0 via npm 6.14.15

### What happened?

Package installation via `npm` fails with error. Installing the same package via npm 8.19.1 or yarn 1.22.19 works fine.

### Steps to reproduce

Run `npm add @pulumi/eks`

### Expected Behavior

Package should be added to the `package.json` dependencies section and the folder `node_modules/@pulumi/eks` should be created.

### Actual Behavior

```
npm ERR! code ENOENT
npm ERR! syscall chmod
npm ERR! path /Users/dan/Dev/scratch-yaml/typescript/node_modules/@pulumi/eks/bin/cmd/provider/index.js
npm ERR! errno -2
npm ERR! enoent ENOENT: no such file or directory, chmod '/Users/dan/Dev/scratch-yaml/typescript/node_modules/@pulumi/eks/bin/cmd/provider/index.js'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent 
```

Relevant lines from debug log:

```
0 info it worked if it ends with ok
1 verbose cli [
1 verbose cli   '/Users/dan/.nodenv/versions/14.18.2/bin/node',
1 verbose cli   '/Users/dan/.nodenv/versions/14.18.2/bin/npm',
1 verbose cli   'add',
1 verbose cli   '@pulumi/eks'
1 verbose cli ]
...
7 http fetch GET 200 https://registry.npmjs.org/@pulumi%2feks 11ms (from cache)
8 http fetch GET 200 https://registry.npmjs.org/@pulumi/eks/-/eks-1.0.0.tgz 10ms (from cache)
9 silly pacote tag manifest for @pulumi/eks@latest fetched in 31ms
...
17 silly resolveWithNewModule @pulumi/eks@1.0.0 checking installable status
...
147 silly decomposeActions fetch @pulumi/eks@1.0.0
148 silly decomposeActions extract @pulumi/eks@1.0.0
149 silly decomposeActions preinstall @pulumi/eks@1.0.0
150 silly decomposeActions build @pulumi/eks@1.0.0
151 silly decomposeActions install @pulumi/eks@1.0.0
152 silly decomposeActions postinstall @pulumi/eks@1.0.0
153 silly decomposeActions finalize @pulumi/eks@1.0.0
154 silly decomposeActions refresh-package-json @pulumi/eks@1.0.0
...
170 silly extract @pulumi/eks@1.0.0
...
180 silly tarball trying @pulumi/eks@latest by hash: sha512-dFHlBDrQtmWkFSNIj1jgkjZga8+l+3JCJakqCJZxdQPwt3ZBu/TqIA2cMvjPWbz3nY/uv35seLJYSqmgDerZmw==
...
190 silly extract @pulumi/eks@latest extracted to /Users/dan/Dev/scratch-yaml/typescript/node_modules/.staging/@pulumi/eks-8241614a (78ms)
...
204 silly finalize /Users/dan/Dev/scratch-yaml/typescript/node_modules/@pulumi/eks/node_modules/@pulumi/aws
205 silly finalize /Users/dan/Dev/scratch-yaml/typescript/node_modules/@pulumi/eks/node_modules/@pulumi/kubernetes
206 silly finalize /Users/dan/Dev/scratch-yaml/typescript/node_modules/@pulumi/eks/node_modules/semver
207 silly finalize /Users/dan/Dev/scratch-yaml/typescript/node_modules/@pulumi/eks/node_modules/which
...
211 silly finalize /Users/dan/Dev/scratch-yaml/typescript/node_modules/@pulumi/eks
...
214 silly refresh-package-json /Users/dan/Dev/scratch-yaml/typescript/node_modules/agent-base
215 silly refresh-package-json /Users/dan/Dev/scratch-yaml/typescript/node_modules/axios/node_modules/form-data
216 silly refresh-package-json /Users/dan/Dev/scratch-yaml/typescript/node_modules/follow-redirects
217 silly refresh-package-json /Users/dan/Dev/scratch-yaml/typescript/node_modules/@pulumi/eks/node_modules/@pulumi/aws
218 silly refresh-package-json /Users/dan/Dev/scratch-yaml/typescript/node_modules/@pulumi/eks/node_modules/@pulumi/kubernetes
219 silly refresh-package-json /Users/dan/Dev/scratch-yaml/typescript/node_modules/@pulumi/eks/node_modules/semver
220 silly refresh-package-json /Users/dan/Dev/scratch-yaml/typescript/node_modules/@pulumi/eks/node_modules/which
221 silly refresh-package-json /Users/dan/Dev/scratch-yaml/typescript/node_modules/axios
222 silly refresh-package-json /Users/dan/Dev/scratch-yaml/typescript/node_modules/https-proxy-agent
223 silly refresh-package-json /Users/dan/Dev/scratch-yaml/typescript/node_modules/netmask
224 silly refresh-package-json /Users/dan/Dev/scratch-yaml/typescript/node_modules/@pulumi/eks
...
247 silly preinstall @pulumi/eks@1.0.0
248 info lifecycle @pulumi/eks@1.0.0~preinstall: @pulumi/eks@1.0.0
249 timing action:preinstall Completed in 1ms
...
260 silly build @pulumi/aws@5.10.0
261 info linkStuff @pulumi/aws@5.10.0
262 silly linkStuff @pulumi/aws@5.10.0 has /Users/dan/Dev/scratch-yaml/typescript/node_modules/@pulumi/eks/node_modules as its parent node_modules
263 silly build @pulumi/kubernetes@3.20.2
264 info linkStuff @pulumi/kubernetes@3.20.2
265 silly linkStuff @pulumi/kubernetes@3.20.2 has /Users/dan/Dev/scratch-yaml/typescript/node_modules/@pulumi/eks/node_modules as its parent node_modules
266 silly build semver@7.3.7
267 info linkStuff semver@7.3.7
268 silly linkStuff semver@7.3.7 has /Users/dan/Dev/scratch-yaml/typescript/node_modules/@pulumi/eks/node_modules as its parent node_modules
269 verbose linkBins [
269 verbose linkBins   { semver: 'bin/semver.js' },
269 verbose linkBins   '/Users/dan/Dev/scratch-yaml/typescript/node_modules/@pulumi/eks/node_modules/.bin',
269 verbose linkBins   false
269 verbose linkBins ]
270 silly gently link link does not already exist {
270 silly gently link   link: '/Users/dan/Dev/scratch-yaml/typescript/node_modules/@pulumi/eks/node_modules/.bin/semver',
270 silly gently link   target: '/Users/dan/Dev/scratch-yaml/typescript/node_modules/@pulumi/eks/node_modules/semver/bin/semver.js'
270 silly gently link }
271 silly build which@1.3.1
272 info linkStuff which@1.3.1
273 silly linkStuff which@1.3.1 has /Users/dan/Dev/scratch-yaml/typescript/node_modules/@pulumi/eks/node_modules as its parent node_modules
274 verbose linkBins [
274 verbose linkBins   { which: 'bin/which' },
274 verbose linkBins   '/Users/dan/Dev/scratch-yaml/typescript/node_modules/@pulumi/eks/node_modules/.bin',
274 verbose linkBins   false
274 verbose linkBins ]
275 silly gently link link does not already exist {
275 silly gently link   link: '/Users/dan/Dev/scratch-yaml/typescript/node_modules/@pulumi/eks/node_modules/.bin/which',
275 silly gently link   target: '/Users/dan/Dev/scratch-yaml/typescript/node_modules/@pulumi/eks/node_modules/which/bin/which'
275 silly gently link }
...
285 silly build @pulumi/eks@1.0.0
286 info linkStuff @pulumi/eks@1.0.0
287 silly linkStuff @pulumi/eks@1.0.0 has /Users/dan/Dev/scratch-yaml/typescript/node_modules as its parent node_modules
288 verbose linkBins [
288 verbose linkBins   { eks: 'bin/cmd/provider/index.js' },
288 verbose linkBins   '/Users/dan/Dev/scratch-yaml/typescript/node_modules/.bin',
288 verbose linkBins   false
288 verbose linkBins ]
289 timing action:build Completed in 3ms
...
294 warn scratch-yaml@ No description
295 warn scratch-yaml@ No repository field.
296 warn scratch-yaml@ No license field.
297 verbose stack Error: ENOENT: no such file or directory, chmod '/Users/dan/Dev/scratch-yaml/typescript/node_modules/@pulumi/eks/bin/cmd/provider/index.js'
298 verbose cwd /Users/dan/Dev/scratch-yaml/typescript
299 verbose Darwin 22.1.0
300 verbose argv "/Users/dan/.nodenv/versions/14.18.2/bin/node" "/Users/dan/.nodenv/versions/14.18.2/bin/npm" "add" "@pulumi/eks"
301 verbose node v14.18.2
302 verbose npm  v6.14.15
303 error code ENOENT
304 error syscall chmod
305 error path /Users/dan/Dev/scratch-yaml/typescript/node_modules/@pulumi/eks/bin/cmd/provider/index.js
306 error errno -2
307 error enoent ENOENT: no such file or directory, chmod '/Users/dan/Dev/scratch-yaml/typescript/node_modules/@pulumi/eks/bin/cmd/provider/index.js'
308 error enoent This is related to npm not being able to find a file.
309 verbose exit [ -2, true ]
```

### Output of `pulumi about`

_No response_

### Additional context

_No response_

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

--- a/Makefile
+++ b/Makefile
@@ -134,7 +134,7 @@ bin/${PROVIDER}:: bin/provider/$(LOCAL_PLAT)/${PROVIDER}
 	cp bin/provider/$(LOCAL_PLAT)/${PROVIDER} bin/${PROVIDER}
 else 
 bin/${PROVIDER}: nodejs/eks/bin nodejs/eks/node_modules
-	cd nodejs/eks && yarn run pkg . ${PKG_ARGS} --target node18 --output $(WORKING_DIR)/bin/${PROVIDER}
+	cd nodejs/eks && yarn run pkg ./bin/cmd/provider/index.js ${PKG_ARGS} --target node18 --output $(WORKING_DIR)/bin/${PROVIDER}
 endif
 
 bin/provider/linux-amd64/${PROVIDER}:: TARGET := node18-linuxstatic-x64
@@ -145,7 +145,7 @@ bin/provider/windows-amd64/${PROVIDER}.exe:: TARGET := node18-win-x64
 bin/provider/%:: nodejs/eks/bin nodejs/eks/node_modules
 	test ${TARGET}
 	cd nodejs/eks && \
-		yarn run pkg . ${PKG_ARGS} --target ${TARGET} --output ${WORKING_DIR}/$@
+		yarn run pkg ./bin/cmd/provider/index.js ${PKG_ARGS} --target ${TARGET} --output ${WORKING_DIR}/$@
 
 dist/${GZIP_PREFIX}-linux-amd64.tar.gz:: bin/provider/linux-amd64/${PROVIDER}
 dist/${GZIP_PREFIX}-linux-arm64.tar.gz:: bin/provider/linux-arm64/${PROVIDER}

--- a/Makefile
+++ b/Makefile
@@ -18,6 +18,7 @@ EKS_SRC 		:= $(wildcard nodejs/eks/*.*) $(wildcard nodejs/eks/*/*.ts) $(wildcard
 LOCAL_PLAT		?= ""
 
 PKG_ARGS 		:= --no-bytecode --public-packages "*" --public
+PKG_TARGET		:= ./bin/cmd/provider/index.js
 
 build:: schema provider build_nodejs build_python build_go build_dotnet build_java
 
@@ -134,7 +135,7 @@ bin/${PROVIDER}:: bin/provider/$(LOCAL_PLAT)/${PROVIDER}
 	cp bin/provider/$(LOCAL_PLAT)/${PROVIDER} bin/${PROVIDER}
 else 
 bin/${PROVIDER}: nodejs/eks/bin nodejs/eks/node_modules
-	cd nodejs/eks && yarn run pkg ./bin/cmd/provider/index.js ${PKG_ARGS} --target node18 --output $(WORKING_DIR)/bin/${PROVIDER}
+	cd nodejs/eks && yarn run pkg ${PKG_TARGET} ${PKG_ARGS} --target node18 --output $(WORKING_DIR)/bin/${PROVIDER}
 endif
 
 bin/provider/linux-amd64/${PROVIDER}:: TARGET := node18-linuxstatic-x64
@@ -145,7 +146,7 @@ bin/provider/windows-amd64/${PROVIDER}.exe:: TARGET := node18-win-x64
 bin/provider/%:: nodejs/eks/bin nodejs/eks/node_modules
 	test ${TARGET}
 	cd nodejs/eks && \
-		yarn run pkg ./bin/cmd/provider/index.js ${PKG_ARGS} --target ${TARGET} --output ${WORKING_DIR}/$@
+		yarn run pkg ${PKG_TARGET} ${PKG_ARGS} --target ${TARGET} --output ${WORKING_DIR}/$@
 
 dist/${GZIP_PREFIX}-linux-amd64.tar.gz:: bin/provider/linux-amd64/${PROVIDER}
 dist/${GZIP_PREFIX}-linux-arm64.tar.gz:: bin/provider/linux-arm64/${PROVIDER}

--- a/nodejs/eks/package.json
+++ b/nodejs/eks/package.json
@@ -3,7 +3,6 @@
     "version": "${VERSION}",
     "description": "Pulumi Amazon Web Services (AWS) EKS Components.",
     "license": "Apache-2.0",
-    "bin": "bin/cmd/provider/index.js",
     "keywords": [
         "pulumi",
         "aws",

